src/Jarvis.Addin.Files/Indexing/FileIndexer.cs

```diff
@@ -131,26 +147,24 @@ private void Index(Trie<IndexedEntry> trie, string word, IndexedEntry entry)
 
         public async Task<IEnumerable<IQueryResult>> Find(string query, CancellationToken token)
         {
-            using (await _lock.ReaderLockAsync(token))
+            var trie = _trie;
```

Comment:
Here, we capture the current `_trie`, so that instead of doing several reads of `_trie` across this method (i.e. so halfway through someone could knock it out from under us), we're *capturing* the current trie, so nobody can knock it out from under us